src/member/mutations.ts

```diff
@@ -107,34 +112,41 @@ export default (queryConfig: QueryClientConfig) => {
     });
   };
 
-  // this mutation is used for its callback and invalidate the keys
   /**
-   * @param {UUID} id parent item id where the file is uploaded in
-   * @param {error} [error] error occurred during the file uploading
+   * Uploads the member profile picture
    */
   const useUploadAvatar = () => {
     const queryClient = useQueryClient();
     return useMutation({
-      mutationFn:
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        async ({ error, data }: { error?: any; data?: any; id: UUID }) => {
-          throwIfArrayContainsErrorOrReturn(data);
-          if (error) throw new Error(JSON.stringify(error));
-        },
+      mutationFn: (args: {
+        file: Blob;
+        onUploadProgress?: (progressEvent: AxiosProgressEvent) => void;
+      }) => {
+        if (args.file.size > MAX_THUMBNAIL_SIZE) {
+          throw new Error(FAILURE_MESSAGES.UPLOAD_BIG_FILES);
+        }
+
+        return Api.uploadAvatar(args, queryConfig);
+      },
       onSuccess: () => {
+        // get memberId from query data
+        const memberId = queryClient.getQueryData<CurrentAccount>(
+          memberKeys.current().content,
+        )?.id;
```

Comment:
Does this work too?

```suggestion
        const { id: memberId } = queryClient.getQueryData<CurrentAccount>(
          memberKeys.current().content,
        );
```

Reply:
No it does not work:
<img width="921" alt="Screenshot 2024-10-09 at 15 00 52" src="https://github.com/user-attachments/assets/5e468581-fdfb-4d72-b537-e6bac7a4f8e3">

Maybe if I add a default, but then I think it is not useful.